src/sharedHooks/useIsConnected.js

```diff
@@ -1,15 +1,29 @@
 // @flow
-
-import { useNetInfo } from "@react-native-community/netinfo";
+import { addEventListener, useNetInfo } from "@react-native-community/netinfo";
+import { useEffect, useState } from "react";
 
 const useIsConnected = ( ): boolean => {
   const netInfo = useNetInfo( );
   // assume there is network connectivity until proven otherwise
   // this should prevent the UI from flickering while isConnected
   // is fetched
-  return netInfo.isConnected === null
-    ? true
-    : netInfo.isConnected;
+  const [isConnected, setIsConnected] = useState(
+    netInfo.isConnected === null
+      ? true
+      : netInfo.isConnected
+  );
+
+  useEffect( () => {
+    const unsubscribe = addEventListener( state => {
+      // Note on ios simulator, these notifications are delayed
+      // https://github.com/react-native-netinfo/react-native-netinfo#issues-with-the-ios-simulator
+      setIsConnected( state.isConnected );
```

Comment:
Might be a bit out of scope, but this and the initial value for `useState` should use `isInternetReachable`. If I'm running a VPN on my phone but the phone is in airplane mode, `isConnected` is true even when `isInternetReachable` is not.